### PR TITLE
Case 3 and case 4

### PR DESCRIPTION
**Pull request by lpoli**
### Changes
This PR is in draft state. It needs to add last two cases of https://github.com/0chain/0chain/issues/2559.
Currently case 3 is add but is yet to be tested. Case 4 is similar to case-3.

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain: https://github.com/0chain/0chain/pull/2675
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/blobbercore/allocation/renamefilechange.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func (rf *RenameFileChange) DeleteTempFile() error {
 	return nil
 }
 
-func (rf *RenameFileChange) ApplyChange(ctx context.Context, rootRef *reference.Ref, change *AllocationChange,
+func (rf *RenameFileChange) applyChange(ctx context.Context, rootRef *reference.Ref, change *AllocationChange,
```
Comment (Hitenjain14) [inline code/go/0chain.net/blobbercore/allocation/renamefilechange.go:27]: Is this change required?

Comment (devyetii) [reply code/go/0chain.net/blobbercore/allocation/renamefilechange.go:27]: We're trying not to re-write the code for conductor tests, that's why we lower-cased the fn. that applies the main logic and kept the upper-case named fn. to be a switch between main flow and conductor testing flow. Is this contradicting with any other code?

**Review by dabasov — APPROVED** [review]
lgtm
